x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/utils/validate.ts

```diff
@@ -95,9 +96,10 @@ export const validateResponseActionsPermissions = async (
 
   const endpointAuthz = await securitySolution.getEndpointAuthz();
 
-  const differences = findDifferenceInArrays<ResponseAction, RuleResponseAction>(
+  const differences = xorWith<ResponseAction | RuleResponseAction>(
```

Comment:
Can it infer the type `ResponseAction | RuleResponseAction` without explicitly providing it?

Reply:
I think it would infer the type if two arrays were of the same types, but in this case I had to specify these.